src/x11/x.c

```diff
@@ -832,7 +832,7 @@ gboolean x_mainloop_fd_prepare(GSource *source, gint *timeout)
         if (timeout)
                 *timeout = -1;
         else
-                g_print("BUG: x_mainloop_fd_prepare: timeout == NULL\n");
+                LOG_E("x_mainloop_fd_prepare: timeout == NULL");
```

Comment:
Glibs docs indicate that the timeout can be NULL but not when or why but I don't think this should be fatal, but should we use `LOG_E` anyways? Not sure about this one.

Reply:
I've never looked at the `GSource` function stuff. For me, this part is a dark chamber, which waits to get lightened up. I'll have a try to explain it to myself.

IIUC, the function should return on the one hand side, a timeout (which will get fed to `poll()` as the timeout) and also should return `TRUE` if the source is already ready.

The [docs say](https://developer.gnome.org/glib/stable/glib-The-Main-Event-Loop.html#GSourceFuncs): 
> Since 2.36 this [timeout_] may be `NULL`, in which case the effect is as if the function always returns `FALSE` with a timeout of `-1`.

The 'effect' is exactly the same like our default. I also don't know, why this may be `NULL`, but we can ignore it, because it's the same like we would have done, if it's not `NULL`.

---

About the `LOG_E` thing here: Regardless if this is only printed to `STDOUT`, just by reading the code (without knowing the actual impact), this smells heavily like a null pointer assertion, which had been gone wrong.